bin/build-plugin-zip.sh

```diff
@@ -7,23 +7,71 @@ set -e
 cd "$(dirname "$0")"
 cd ..
 
+# Make sure there are no changes in the working tree
+changed=
+if ! git diff --exit-code > /dev/null; then
+	changed="file(s) modified"
+elif ! git diff --cached --exit-code > /dev/null; then
+	changed="file(s) staged"
+fi
+if [ ! -z "$changed" ]; then
+	git status
+	echo "ERROR: Cannot build plugin zip with dirty working tree."
+	echo "       Commit your changes and try again."
+	exit 1
+fi
+
+branch="$(git rev-parse --abbrev-ref HEAD)"
+if [ "$branch" != 'master' ]; then
+	echo "WARNING: You should probably be running this script against the"
+	echo "         'master' branch (current: '$branch')"
+	echo
+	sleep 2
+fi
+
+# Download all vendor scripts
+vendor_scripts=""
+exec 3< <(
+	# minified versions of vendor scripts
+	php bin/get-vendor-scripts.php
+	# non-minified versions of vendor scripts (for SCRIPT_DEBUG)
+	php bin/get-vendor-scripts.php debug
+)
+while IFS='|' read -u 3 url filename; do
+	wget -nv "$url" -O "vendor/$$.tmp.js"
```

Comment:
The issues with `$$` and other "weak" methods of temporary filename generation are related to privileged processes that can be tricked into writing unexpected files that a user would not otherwise be able to write to.  Thankfully, this set of issues does not apply to us.

Still, I've removed the use of `$$` to avoid promoting this practice.